p/scripts/main.js

```diff
@@ -1087,8 +1087,19 @@ function init_stream(stream) {
 		}
 
 		el = ev.target.closest('.item.share > button[data-type="clipboard"]');
-		if (el && navigator.clipboard) {	// Clipboard
-			navigator.clipboard.writeText(el.dataset.url);
+		if (el) { // Clipboard
+			if (navigator.clipboard) {
+				navigator.clipboard.writeText(el.dataset.url);
+			} else {
```

Comment:
```suggestion
			} else if (document.execCommand) {
```

`document.execCommand` seems deprecated https://developer.mozilla.org/docs/Web/API/document/execCommand

Reply:
yes it is deprecated. But how could be the user informed, if it is not available? To have a "Clipboard" link without any positive result is not fine.

Reply:
Yes, this PR is simply what I said in https://github.com/FreshRSS/FreshRSS/issues/5595#issuecomment-1685238478

The scare phrasing on MDN doesn't reflect reality though. Most of what it does doesn't have a replacement yet (though copy does) and bugs are still being actively fixed in all browsers.

Reply:
> But how could be the user informed, if it is not available

Maybe simply an `alert()`? In any case, it is better to test than having a JavaScript error.

Reply:
As an aside, if you're worried about stuff like that note that the `navigator.clipboard.writeText()` promise can fail and that `document.execCommand('copy')` can return false.

Reply:
@Alkarex 
> In any case, it is better to test than having a JavaScript error.

What JS error are you referring to? It's already testing if document.execCommand exists. If copy fails it returns false.

Reply:
> What JS error are you referring to? It's already testing if document.execCommand exists.

The `if (document.execCommand)` is my suggestion

Reply:
Oops, indeed it is.

Though while I do think it's good practice, it exist on basically any browser from the past 20 years, and there's pretty much no way they can actually drop it either. If it's gone anytime before the next decade I'll, uh, buy you a coffee or something (I don't want to eat my hat on the off chance it does happen). ;-D